osrparse/dump.py

```diff
@@ -0,0 +1,102 @@
+import lzma
+import struct
+
+from osrparse.utils import (ReplayEventOsu, ReplayEventTaiko, ReplayEventCatch,
+    ReplayEventMania)
+from osrparse.replay import Replay
+
+def pack_byte(data: int):
+    return struct.pack("<B", data)
+
+def pack_short(data: int):
+    return struct.pack("<H", data)
+
+def pack_int(data: int):
+    return struct.pack("<I", data)
+
+def pack_long(data: int):
+    return struct.pack("<Q", data)
+
+def pack_ULEB128(data):
+    # taken from https://github.com/mohanson/leb128
+    r, i = [], len(data)
+
+    while True:
+        byte = i & 0x7f
+        i = i >> 7
+
+        if (i == 0 and byte & 0x40 == 0) or (i == -1 and byte & 0x40 != 0):
+            r.append(byte)
+            return b"".join(map(pack_byte, r))
+
+        r.append(0x80 | byte)
+
+def pack_string(data: str):
+    if data:
+        return pack_byte(11) + pack_ULEB128(data) + data.encode("utf-8")
+    return pack_byte(11) + pack_byte(0)
+
+def dump_timestamp(replay):
+    # due to converting ticks into an int, there appears to be some
+    # precision loss in the ticks value even though it's small (around 4
+    # ticks), it might be an issue with reproccessing the same replay for
+    # many times in a row
+
+    # when it's negative, it represents January 1st 0001, 01:00:00 AM UTC
+    shift = 62135596800 + 3600
```

Comment:
what's the significance of 62135596800?

Reply:
osu! stores the timestamp for a replay as Windows ticks, which is basically time in seconds that have passed since January 1st 0001 at midnight in UTC time.
the number is basically the absolute value from the date above in a Unix timestamp
by adding the value to a regular Unix timestamp, it will result in Windows time ticks, but in seconds instead of nanoseconds
from there you can multiply it and get a vaild timestamp for the replay

Reply:
so windows ticks starts at year 0001 instead of 1970...weird. The "negative" comment here was actually coming from your timezone (which is UTC +1) since `timestamp` wasn't timezone-aware; I've fixed this in e863a91c281ad46646fc1bd573a7609b2852a444.